ksql-common/src/main/java/io/confluent/ksql/schema/ksql/SqlTypeWalker.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.schema.ksql;
+
+import com.google.common.collect.ImmutableMap;
+import io.confluent.ksql.schema.ksql.types.SqlArray;
+import io.confluent.ksql.schema.ksql.types.SqlDecimal;
+import io.confluent.ksql.schema.ksql.types.SqlMap;
+import io.confluent.ksql.schema.ksql.types.SqlPrimitiveType;
+import io.confluent.ksql.schema.ksql.types.SqlStruct;
+import io.confluent.ksql.schema.ksql.types.SqlStruct.Field;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+/**
+ * Visitor pattern for ksql types.
+ */
+public final class SqlTypeWalker {
```

Comment:
note: the equivalent of `SqlSchemaWalker` for the connect schema. However, this time we only need to worry about the specific types KSQL uses, not the full Connect set of types.

Reply:
Is this class actually being used anywhere yet?

Reply:
Not yet. Adding as it'll be needed later.